Java/core/src/main/java/com/amazon/randomcutforest/tree/AbstractRandomCutTree.java

```diff
@@ -270,75 +270,99 @@ void updateAncestorNodesAfterDelete(NodeReference nodeReference, Point point) {
         updateAncestorPointSum(nodeReference);
     }
 
+    NodeReference findLeaf(Point point, boolean deleteMass) {
+        NodeReference nodeReference = root;
+        while (!isLeaf(nodeReference)) {
+            if (deleteMass) {
+                decrementMass(nodeReference);
+            }
+            nodeReference = (leftOf(point, getCutDimension(nodeReference), getCutValue(nodeReference)))
+                    ? getLeftChild(nodeReference)
+                    : getRightChild(nodeReference);
+        }
+        return nodeReference;
+    }
+
+    NodeReference findLeafAndVerify(Point point, boolean deleteMass) {
+        NodeReference nodeReference = findLeaf(point, deleteMass);
+        Point oldPoint = getPointFromLeafNode(nodeReference);
+        if (!equals(oldPoint, point)) {
+            throw new IllegalStateException(toString(point) + " " + toString(getPointFromLeafNode(nodeReference)) + " "
+                    + nodeReference + " node " + false + " Inconsistency in trees in delete step here.");
+        }
+        return nodeReference;
+    }
+
     /**
-     * method to delete a point from the tree
+     * the following function returns the number of copies of a point in the tree
+     * and switches the reference to the provided reference. This may be useful for
+     * collating duplicate points across trees. By default this is set to no-op.
```

Comment:
The part in the comment about the default isn't true, because there's no default implementation of the method.

Reply:
fixed.